src/common/support_utils.cpp

```diff
@@ -15,65 +15,69 @@ char *eofstr(char *str) {
     return (str + strlen(str));
 }
 
-void block2hex(char *str, uint8_t *pdata, size_t length) {
+void block2hex(char *str, uint32_t str_size, uint8_t *pdata, size_t length) {
     for (; length > 0; length--)
-        sprintf(eofstr(str), "%02X", *(pdata++));
+        snprintf(eofstr(str), str_size - strlen(str), "%02X", *(pdata++));
```

Comment:
Since snprintf actually returns the number of characters, we may avoid computing strlen at all. Moreover when we do it twice now for obvious reasons. I doubt the compiler can optimize the strlen calls into a single one.